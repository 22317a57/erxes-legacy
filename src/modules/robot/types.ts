export type IFeature = {
  name: string;
  text: string;
};
