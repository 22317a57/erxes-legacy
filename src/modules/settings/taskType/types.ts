export interface ITaskType {
  _id: string;
  name: string;
  icon: string;
}

// query types

export type TaskTypeQueryResponse = {
  taskTypes: ITaskType[];
  loading: boolean;
  refetch: () => void;
};

export type MutationVariables = {
  _id?: string;
  name?: string;
  icon?: string;
};

// mutation types

export type AddMutationResponse = {
  addMutation: (mutation: { variables: MutationVariables }) => Promise<any>;
};

export type EditMutationResponse = {
  editMutation: (mutation: { variables: MutationVariables }) => Promise<any>;
};

export type RemoveMutationResponse = {
  removeMutation: (mutation: { variables: { _id: string } }) => Promise<any>;
};
