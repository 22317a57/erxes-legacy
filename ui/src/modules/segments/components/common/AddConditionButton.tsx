import Button from 'modules/common/components/Button';
import Icon from 'modules/common/components/Icon';
import { __ } from 'modules/common/utils';
import React from 'react';
import OverlayTrigger from 'react-bootstrap/OverlayTrigger';
import Popover from 'react-bootstrap/Popover';
import { ISegmentCondition, ISegmentField } from '../../types';
import { FieldType, PopoverList } from '../styles';

type Props = {
  fields: ISegmentField[];
  events: string[];
  addCondition: (condition: ISegmentCondition) => void;
};

class AddConditionButton extends React.Component<Props> {
  private overlayTrigger;

  addCondition = (type: string) => {
    this.props.addCondition({
      key: Math.random().toString(),
      propertyName: '',
      propertyValue: '',
      propertyOperator: '',
      type,
    });

    this.overlayTrigger.hide();
  };

  renderPopover() {
    const addPropertyCondition = () => {
      return this.addCondition('property');
    }

    const addEventCondition = () => {
      return this.addCondition('event');
    }

    return (
      <Popover id="condition-popover">
        <Popover.Title as="h3">{__('Select a field')}</Popover.Title>
        <Popover.Content>
          <PopoverList onClick={addPropertyCondition}>
            <FieldType>
              {__("Fields")}
              <Icon icon="information" />
            </FieldType>
          </PopoverList>

          <PopoverList onClick={addEventCondition}>
            <FieldType>
              {__("Events")}
              <Icon icon="settings-3" />
            </FieldType>
          </PopoverList>
        </Popover.Content>
      </Popover>
    );
  }

  render() {
    return (
      <OverlayTrigger
        ref={overlayTrigger => {
          this.overlayTrigger = overlayTrigger;
        }}
        trigger="click"
        placement="bottom"
        overlay={this.renderPopover()}
        container={this}
        rootClose={true}
      >
        <Button btnStyle="success" icon="add">
          Add a condition
        </Button>
      </OverlayTrigger>
    );
  }
}

export default AddConditionButton;
