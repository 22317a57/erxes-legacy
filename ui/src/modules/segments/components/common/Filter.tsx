import Button from "modules/common/components/Button";
import { FormControl } from "modules/common/components/form";
import { __ } from "modules/common/utils";
import { operators } from "modules/customers/constants";
import { FlexContent, FlexItem, FlexRightItem } from "modules/layout/styles";
import React from "react";
import { IConditionFilter } from "../../types";

type Props = {
  names: string[];
  filter: IConditionFilter;
  onChange: (filter: IConditionFilter) => void;
  onRemove?: (id: string) => void;
};

type State = {
  key: string;
  currentName: string;
  currentOperator: string;
  currentValue: string;
};

class Filter extends React.Component<Props, State> {
  constructor(props) {
    super(props);

    const { filter } = this.props;

    this.state = {
      key: filter.key || '',
      currentName: filter.name,
      currentOperator: filter.operator,
      currentValue: filter.value,
    };
  }

  onChange = () => {
    const { currentName, currentOperator, currentValue } = this.state;
    const { onChange, filter } = this.props;

    return onChange({
      key: filter.key,
      name: currentName,
      operator: currentOperator,
      value: currentValue,
    });
  }

  onChangeValue = (e: React.FormEvent<HTMLElement>) => {
    this.setState({ currentValue: (e.currentTarget as HTMLInputElement).value }, this.onChange);
  }

  onChangeNames = (e: React.FormEvent<HTMLElement>) => {
    this.setState({ currentName: (e.currentTarget as HTMLInputElement).value }, this.onChange);
  }

  onChangeOperators = (e: React.FormEvent<HTMLElement>) => {
    this.setState({ currentOperator: (e.currentTarget as HTMLInputElement).value }, this.onChange);
  }

  renderNames() {
    const { names } = this.props;
    const { currentName } = this.state;

    return (
      <FormControl componentClass="select" placeholder={__("select")} onChange={this.onChangeNames} value={currentName}>
        <option />
        {names.map((name, index) => (
          <option value={name} key={index}>
            {name}
          </option>
        ))}
      </FormControl>
    );
  }

  renderOperators() {
    const { currentOperator } = this.state;

    return (
      <FormControl componentClass="select" placeholder={__("select")} onChange={this.onChangeOperators} value={currentOperator}>
        <option />
        {operators.map(c => (
          <option value={c.value} key={c.value}>
            {c.name}
          </option>
        ))}
      </FormControl>
    );
  }

  onRemove = () => {
    const { onRemove } = this.props;

    if (onRemove) {
      onRemove(this.props.filter.key || '');
    }
  };

  renderRemoveButton = () => {
    const { onRemove } = this.props;

    if (!onRemove) {
      return;
    }

    return <Button btnStyle="danger" size="small" icon="cancel-1" onClick={this.onRemove} />;
  }

  render() {
    const { currentValue } = this.state;

    return (
      <FlexContent>
        <FlexItem>
          {this.renderNames()}
          {this.renderOperators()}
          <input value={currentValue} onChange={this.onChangeValue} />
        </FlexItem>
        <FlexRightItem>
          {this.renderRemoveButton()}
        </FlexRightItem>
      </FlexContent>
    );
  }
}

export default Filter;
