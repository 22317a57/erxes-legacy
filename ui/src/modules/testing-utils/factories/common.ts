import * as Factory from 'factory.ts';
import { IConditionsRule } from 'modules/common/types';

export const conditionsRuleFactory = Factory.Sync.makeFactory<IConditionsRule>({
  _id: '1',
  kind: 'string',
  text: 'string',
  condition: 'string',
  value: 'string',
});