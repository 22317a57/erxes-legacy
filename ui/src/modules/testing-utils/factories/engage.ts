import * as Factory from 'factory.ts';
import { IAttachment } from 'modules/common/types';
import { IEmailDelivery, IEngageEmail, IEngageMessage, IEngageMessenger, IEngageScheduleDate, IEngageStats } from 'modules/engage/types';
import { ISegment, ISegmentCondition } from 'modules/segments/types';
import { IEmailSignature } from 'modules/settings/email/types';
import { brandFactory } from './brands';
import { conditionsRuleFactory } from './common';
import { tagFactory } from './tags';
import { userFactory } from './user';

export const engageMessengerFactory = Factory.Sync.makeFactory<IEngageMessenger>({
  brandId: '22',
  kind: 'auto',
  sentAs: 'Dorj',
  content: 'Hello',
  rules: [
    conditionsRuleFactory.build(),
    conditionsRuleFactory.build({ _id: '21' })
  ]
});

export const attachmentFactory = Factory.Sync.makeFactory<IAttachment>({
  name: 'Images1',
  type: 'img',
  url: '/images/logo',
  size: 12
});

export const engageEmailFactory = Factory.Sync.makeFactory<IEngageEmail>({
  templateId: '3',
  subject: 'Subject',
  content: 'content',
  attachments: [
    attachmentFactory.build(),
    attachmentFactory.build({ name: 'Images1' })
  ],
});

export const segmentConditionFactory = Factory.Sync.makeFactory<ISegmentCondition>({
  _id: '15',
  field: 'auto',
  value: 'string',
  operator: 'string',
  dateUnit: 'string',
  type: 'string',
  brandId: 'string',
});

export const segmentFactory = Factory.Sync.makeFactory<ISegment>({
  _id: '5',
  contentType: 'type',
  getSubSegments: [],
  getParentSegment: {} as ISegment,
  // ISegmentDocs
  name: 'segment',
  description: 'desc',
  color: '#fff',
  connector: 'connect',
  conditions: [
    segmentConditionFactory.build(),
    segmentConditionFactory.build({ _id: '15' })
  ],
  subOf: 'sub'
});

export const emailSignatureFactory = Factory.Sync.makeFactory<IEmailSignature>({
  brandId: '22',
  signature: 'erxes'
});

export const engageStatsFactory = Factory.Sync.makeFactory<IEngageStats>({
  send: 3,
  delivery: 4,
  open: 5,
  click: 11,
  complaint: 1,
  bounce: 6,
  renderingfailure: 7,
  reject: 9,
  total: 25,
});

export const engageScheduleDateFactory = Factory.Sync.makeFactory<IEngageScheduleDate>({
  type: 'everyday',
  month: 'July',
  day: '23',
  time: new Date()
});

export const engageMessageFactory = Factory.Sync.makeFactory<IEngageMessage>({
  _id: '4',
  stopDate: new Date(),
  createdDate: new Date(),
  messengerReceivedCustomerIds: ['test', '4'],
  brand: brandFactory.build({ _id: '3', name: 'Erxes' }),
  segment: segmentFactory.build({ _id: '5' }),
  fromUser: userFactory.build({ _id: '12', email: 'erxes@nmma.co' }),
  tagIds: ['12', '13'],
  getTags: [
    tagFactory.build(),
    tagFactory.build({ _id: '23' })
  ],
  stats: engageStatsFactory.build({}),
  logs: [{ message: 'string' }],
  // IEngageMessageDocs
  kind: 'auto',
  type: 'email',
  segmentIds: ['1,2'],
  brandIds: ['3,4'],
  customerIds: ['3,4'],
  title: 'test',
  fromUserId: '22',
  method: '22',
  isDraft: true,
  isLive: false,
  email: engageEmailFactory.build({ templateId: '3' }),
  messenger: engageMessengerFactory.build({ brandId: '22' }),
  scheduleDate: engageScheduleDateFactory.build({ type: 'everyday', month: 'July', day: '12', time: '2' })
});

export const emailDeliveryFactory = Factory.Sync.makeFactory<IEmailDelivery>({
  _id: '33',
  subject: 'subject',
  body: 'Body',
  to: 'Dulam@yahoo.com',
  cc: 'erxes@gmail.com',
  bcc: 'erxes@nmma.co',
  attachments: [JSON],
  from: 'erxes',
  kind: 'auto',
  userId: '12',
  customerId: '22',

  fromUser: userFactory.build({ _id: '12' }),
  fromEmail: 'erxes@nmma.co'
});