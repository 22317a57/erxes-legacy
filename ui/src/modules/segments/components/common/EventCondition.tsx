import Button from 'modules/common/components/Button';
import { FormControl } from 'modules/common/components/form';
import { __ } from 'modules/common/utils';
import { FlexContent, FlexItem, FlexRightItem } from 'modules/layout/styles';
import React from 'react';
import { IConditionFilter } from '../../types';
import { ConditionItem } from '../styles';
import Filter from './Filter';

type Props = {
  events: string[];
  conditionKey: string;
  name: string;
  attributeNames: string[];
  attributeFilters: IConditionFilter[];
  onChange: (args: { key: string, name: string, attributeFilters: IConditionFilter[] }) => void;
  onRemove: (id: string) => void;
};

type State = {
  attributeFilters: IConditionFilter[];
  currentEvent?: string;
}

class Condition extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);

    this.state = {
      currentEvent: props.name,
      attributeFilters: props.attributeFilters.map(filter => ({ key: Math.random().toString(), ...filter }))
    }
  }

  removeCondition = () => {
    this.props.onRemove(this.props.conditionKey);
  };

  onChangeFilter = () => {
    const { onChange, conditionKey } = this.props;
    const { currentEvent, attributeFilters } = this.state;

    return onChange({
      key: conditionKey,
      name: currentEvent || '',
      attributeFilters,
    });
  }

  onChangeAttributeFilter = (filter: IConditionFilter) => {
    this.setState({
      attributeFilters: this.state.attributeFilters.map(f =>
        f.key === filter.key ? filter : f
      )
    }, this.onChangeFilter);
  };

  onRemoveAttributeFilter = (key: string) => {
    const attributeFilters = this.state.attributeFilters.filter(f => f.key !== key);

    this.setState({ attributeFilters }, this.onChangeFilter);
  };

  onChangeEvents = (e) => {
    this.setState({ currentEvent: e.currentTarget.value }, this.onChangeFilter);
  }

  addAttributeFilter = () => {
    const attributeFilter = {
      key: Math.random().toString(),
      name: '',
      operator: '',
      value: ''
    };

    const { attributeFilters } = this.state;

    attributeFilters.push(attributeFilter);

    this.setState({ attributeFilters });
  }

  renderAttributeFilters = () => {
    const { attributeFilters } = this.state;
    const { attributeNames } = this.props;

    return attributeFilters.map((filter, index) => {
      return (
        <Filter
          key={index}
          names={attributeNames}
          filter={filter}
          onChange={this.onChangeAttributeFilter}
          onRemove={this.onRemoveAttributeFilter}
        />
      );
    });
  }

  renderNames() {
    const { events } = this.props;
    const { currentEvent } = this.state;

    return (
      <FormControl componentClass="select" placeholder={__("select")} onChange={this.onChangeEvents} value={currentEvent}>
        <option />

        {events.map((name, index) => (
          <option value={name} key={index}>
            {name}
          </option>
        ))}
      </FormControl>
    );
  }

  render() {
    return (
      <ConditionItem>
        <FlexContent>
          <FlexItem>
            {this.renderNames()}

            <div>
              {this.renderAttributeFilters()}
            </div>

            <div>
              <Button onClick={this.addAttributeFilter}>
                Add attribute
              </Button>
            </div>
          </FlexItem>

          <FlexRightItem>
            <Button
              btnStyle="danger"
              size="small"
              icon="cancel-1"
              onClick={this.removeCondition}
            />
          </FlexRightItem>
        </FlexContent>
      </ConditionItem>
    );
  }
}

export default Condition;
