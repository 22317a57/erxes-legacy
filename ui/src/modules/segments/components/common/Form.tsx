import Button from 'modules/common/components/Button';
import FormControl from 'modules/common/components/form/Control';
import CommonForm from 'modules/common/components/form/Form';
import FormGroup from 'modules/common/components/form/Group';
import ControlLabel from 'modules/common/components/form/Label';
import { IButtonMutateProps, IFormProps } from 'modules/common/types';
import { __, generateRandomColorCode } from 'modules/common/utils';
import { FlexContent, FlexItem } from 'modules/layout/styles';
import {
  IConditionFilter,
  IEvent,
  ISegment,
  ISegmentCondition,
  ISegmentWithConditionDoc
} from 'modules/segments/types';
import React from 'react';
import { Link } from 'react-router-dom';
import { ConditionWrapper, SegmentTitle, SegmentWrapper } from '../styles';
import AddConditionButton from './AddConditionButton';
import EventCondition from './EventCondition';
import PropertyCondition from './PropertyCondition';

type Props = {
  contentType?: string;
  fields: any[];
  events: IEvent[];
  renderButton: (props: IButtonMutateProps) => JSX.Element;
  edit?: (params: { _id: string; doc: ISegmentWithConditionDoc }) => void;
  segment?: ISegment;
  headSegments: ISegment[];
  count: (segment: ISegment) => void;
  isForm?: boolean;
  afterSave?: () => void;
};

type State = {
  name: string;
  description: string;
  subOf: string;
  color: string;
  conditions: ISegmentCondition[];
};

class Form extends React.Component<Props, State> {
  constructor(props) {
    super(props);

    const segment: ISegment = props.segment || {
      name: '',
      description: '',
      subOf: '',
      color: generateRandomColorCode(),
      conditions: [],
    };

    segment.conditions = segment.conditions.map(
      (cond: ISegmentCondition) => ({
        key: Math.random().toString(),
        ...cond
      })
    );

    this.state = segment;
  }

  addCondition = (condition: ISegmentCondition) => {
    this.setState({
      conditions: [...this.state.conditions, condition]
    });
  };

  changeEventCondition = (args: { key: string, name: string, attributeFilters: IConditionFilter[] }) => {
    const condition = {
      type: 'event',
      key: args.key,
      eventName: args.name,
      eventAttributeFilters: (args.attributeFilters || []).map(filter => {
        const { key, ...rest } = filter;

        return rest;
      })
    }

    this.setState({
      conditions: this.state.conditions.map(c =>
        c.key === condition.key ? condition : c
      )
    });
  };

  changePropertyCondition = (args: { key: string, name: string, operator: string, value: string }) => {
    const condition = {
      type: 'property',
      key: args.key,
      propertyName: args.name,
      propertyOperator: args.operator,
      propertyValue: args.value,
    }

    this.setState({
      conditions: this.state.conditions.map(c =>
        c.key === condition.key ? condition : c
      )
    });
  };

  removeCondition = (key: string) => {
    const conditions = this.state.conditions.filter(c => c.key !== key);

    this.setState({ conditions });
  };

  handleChange = <T extends keyof State>(name: T, value: State[T]) => {
    this.setState({ [name]: value } as Pick<State, keyof State>);
  };

  generateDoc = (values: {
    _id?: string;
    name: string;
    subOf: string;
    color: string;
  }) => {
    const { segment, contentType } = this.props;
    const { conditions } = this.state;
    const finalValues = values;

    const updatedConditions: ISegmentCondition[] = [];

    if (segment) {
      finalValues._id = segment._id;
    }

    conditions.forEach((cond: ISegmentCondition) => {
      const { key, ...rest } = cond;
      updatedConditions.push(rest);
    });

    return {
      ...finalValues,
      contentType,
      conditions: updatedConditions
    };
  };

  renderParent() {
    const { contentType } = this.props;
    const { subOf } = this.state;

    if (!subOf) {
      return null;
    }

    return (
      <React.Fragment>
        <Link
          to={`/segments/edit/${contentType}/${subOf}`}
          target="_blank"
        >
          <Button icon="eye" ignoreTrans={true}>
            {__('Parent segment conditions')}
          </Button>
        </Link>
        <hr />
      </React.Fragment>
    );
  }

  renderCondition(condition: ISegmentCondition) {
    const { fields, events } = this.props;

    if (condition.type === 'property') {
      return (
        <PropertyCondition
          fields={fields}
          key={condition.key}
          conditionKey={condition.key || ''}
          name={condition.propertyName || ''}
          operator={condition.propertyOperator || ''}
          value={condition.propertyValue || ''}
          onChange={this.changePropertyCondition}
          onRemove={this.removeCondition}
        />
      )
    }

    return (
      <EventCondition
        events={events}
        key={condition.key}
        conditionKey={condition.key || ''}
        name={condition.eventName || ''}
        attributeFilters={condition.eventAttributeFilters || []}
        onChange={this.changeEventCondition}
        onRemove={this.removeCondition}
      />
    )
  }

  renderConditions() {
    const { conditions } = this.state;

    return (
      <React.Fragment>
        <ConditionWrapper>
          {this.renderParent()}
          {conditions.map(condition => this.renderCondition(condition))}
          </ConditionWrapper>
        <AddConditionButton addCondition={this.addCondition} />
      </React.Fragment>
    );
  }

  renderSubOf(formProps: IFormProps) {
    const onChange = (e: React.FormEvent) =>
      this.handleChange('subOf', (e.currentTarget as HTMLInputElement).value);

    return (
      <FormGroup>
        <ControlLabel>Sub segment of</ControlLabel>
        <FormControl
          {...formProps}
          name="subOf"
          componentClass="select"
          value={this.state.subOf || ''}
          onChange={onChange}
        >
          <option value="">[not selected]</option>
          {this.props.headSegments.map(segment => (
            <option value={segment._id} key={segment._id}>
              {segment.name}
            </option>
          ))}
        </FormControl>
      </FormGroup>
    );
  }

  renderForm = (formProps: IFormProps) => {
    const {
      isForm,
      segment,
      contentType,
      renderButton,
      afterSave
    } = this.props;

    const { values, isSubmitted } = formProps;
    const { name, description, color } = this.state;

    const nameOnChange = (e: React.FormEvent) =>
      this.handleChange('name', (e.currentTarget as HTMLInputElement).value);

    const descOnChange = (e: React.FormEvent) =>
      this.handleChange(
        'description',
        (e.currentTarget as HTMLInputElement).value
      );

    const colorOnChange = (e: React.FormEvent) =>
      this.handleChange('color', (e.currentTarget as HTMLInputElement).value);

    return (
      <>
        <FormGroup>
          <ControlLabel required={true}>Name</ControlLabel>
          <FormControl
            {...formProps}
            name="name"
            value={name}
            onChange={nameOnChange}
            required={true}
            autoFocus={true}
          />
        </FormGroup>
        <FormGroup>
          <ControlLabel>Description</ControlLabel>
          <FormControl
            {...formProps}
            name="description"
            value={description}
            onChange={descOnChange}
          />
        </FormGroup>
        {this.renderSubOf({ ...formProps })}
        <FormGroup>
          <ControlLabel>Color</ControlLabel>
          <FormControl
            {...formProps}
            name="color"
            type="color"
            value={color}
            onChange={colorOnChange}
          />
        </FormGroup>
        <Button.Group>
          {isForm && (
            <Link to={`/segments/${contentType}`}>
              <Button size="small" btnStyle="simple" icon="cancel-1">
                Cancel
              </Button>
            </Link>
          )}

          {renderButton({
            name: 'segment',
            values: this.generateDoc(values),
            callback: afterSave,
            isSubmitted,
            object: segment
          })}
        </Button.Group>
      </>
    );
  };

  render() {
    return (
      <SegmentWrapper>
        <SegmentTitle>{__('Filters')}</SegmentTitle>
        {this.renderConditions()}
        <hr />
        <FlexContent>
          <FlexItem count={4}>
            <CommonForm renderContent={this.renderForm} />
          </FlexItem>
          <FlexItem count={2} />
        </FlexContent>
      </SegmentWrapper>
    );
  }
}

export default Form;
