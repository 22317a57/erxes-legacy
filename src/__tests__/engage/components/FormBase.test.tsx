import { mount, shallow } from 'enzyme';
import { IBreadCrumbItem } from 'modules/common/types';
import FormBase from 'modules/engage/components/FormBase';
import { IEngageMessageDoc } from 'modules/engage/types';
import * as React from 'react';
import * as renderer from 'react-test-renderer';

describe('FormBase component', () => {
  const defaultProps = {
    kind: 'string',
    content: (params: {
      renderTitle: () => string;
      breadcrumbs: IBreadCrumbItem[];
      validateDoc: (
        type: string,
        doc: IEngageMessageDoc
      ) => { status: string; doc?: IEngageMessageDoc };
    }) => 'any'
  };

  test('renders successfully', () => {
    shallow(<FormBase {...defaultProps} />);
  });

  test('renders with default props', () => {
    const control = mount(<FormBase {...defaultProps} />);
    const props = control.props();

    expect(props).toMatchObject(defaultProps);
  });

  test('snapshot matches', () => {
    const rendered = renderer.create(<FormBase {...defaultProps} />).toJSON();

    expect(rendered).toMatchSnapshot();
  });
});
