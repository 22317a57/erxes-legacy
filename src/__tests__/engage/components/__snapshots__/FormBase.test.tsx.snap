// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormBase component snapshot matches 1`] = `"any"`;
